### PR TITLE
mercurial support

### PR DESCRIPTION
**Pull request by bergholdt**
Added support for mercurial.

Some of us have still not moved all our repositories to git, so support for other version control systems are useful.

### CODE REVIEW

**Comment by bergholdt** [comment]
Sorry I did not make test.

I'm developing on cooperate windows machine where all current specs fails.

File: bump.gemspec [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Gem::Specification.new "bump" do |s|
-  s.version = "0.5.3"
+  s.version = "0.5.4"
```
Comment (grosser) [inline bump.gemspec]: never bump versions in PRs

File: lib/bump.rb [hunk 2]
```diff
@@ -180,9 +173,7 @@ def self.next_version(current, part)
...
     end
 
-    def self.under_version_control?(file)
-      @all_files ||= `git ls-files`.split(/\r?\n/)
-      @all_files.include?(file)
-    end
   end
 end
+
+require 'bump/version_control'
```
Comment (grosser) [inline lib/bump.rb]: - requires go to the top of the file
 - always add trailing newline

File: lib/bump/version_control.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+module Bump
+  class VersionControlNotFoundError < StandardError; end
+
+  class VersionControl
+    def self.git?
```
Comment (grosser) [inline lib/bump/version_control.rb]: use class << self + private to make only the methods public that you want to use from the outside

File: lib/bump/version_control.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    def self.commit_message(version, options)
+      (options[:commit_message]) ? "v#{version} #{options[:commit_message]}" : "v#{version}"
+    end
+
+    def self.under_version_control?(file)
+      case
+      when git?
+        @all_files ||= `git ls-files`.split(/\r?\n/)
```
Comment (grosser) [inline lib/bump/version_control.rb]: can use `@all_files ||= case` to avoid the duplication

**Comment by grosser** [comment]
don't develop on windows (use docker / ubuntu etc) or fix the tests then :)

File: lib/bump/version_control.rb (hunk 3 above)
Comment (bergholdt) [inline lib/bump/version_control.rb]: Did not think there was any harm in giving access to all methods as they might be useful. And the one not currently not used outside does not change anything (git?, mercurial?, commit_message).

Will make them private if you wish so.

Comment (grosser) [inline lib/bump/version_control.rb]: I try to when in doubt make things private, if we later change/move that then nobody will be surprised that these methods went away

Comment (bergholdt) [inline lib/bump/version_control.rb]: OK you are right - I made them private.

**Comment by gregorym** [comment]
Can you please rebase.

**Comment by gregorym** [comment]
Can you also add a section to the README with all the supported version control systems.

**Comment by bergholdt** [comment]
Travis build fails on `git status`.should include "nothing to commit" on ruby 2.1.6 and 2.2.3 - This also on PR #53 

I can reproduce this local on bundle 1.12.1 but not on 1.11.2 (https://github.com/bundler/bundler/blob/master/CHANGELOG.md)
Solution - add .bundle/ to .gitignore

**Pull request closed without merging**